src/formatters/mdapi/retrieveResultFormatter.ts

```diff
@@ -51,8 +51,7 @@ export class RetrieveResultFormatter extends ResultFormatter {
     this.result = Object.assign({}, result.response, { zipFilePath });
```

Comment:
if `result` is not guaranteed, you'd need a nullco here, too.  Or maybe you don't need it in the `this.warnings`.